File: recursion.cpp
#include <iostream>
using namespace std;

int pow(int base, int exp)
{
    return (exp > 0) ? base * pow(base, exp - 1) : 1;
}

int main(void)
{
    int base, exp;
        cout << "Enter the base number: ";
        cin >> base;
    loop:
        cout << "Enter the power of the number: ";
        cin >> exp;
        if (exp < 0)
        {
            cout << "OutOfBounds: Only positive exponents are accepted." << endl;
            goto loop;
        }
    cout << base << "^" << exp << " = " << pow(base, exp);
}

File: recursion.py
def pow(base, exp):
    if exp > 0:
        return base * pow(base, exp - 1)
    elif exp == 0:
        return 1

base = int(input("Enter the base number: "))
exp = int(input("Enter the power of the number: "))
print(base, "^", exp, " = ", pow(base,exp), sep = '')